Automattic/woocommerce-payments
# Add BNPL to cart summary

Fixes #8144 

#### Changes proposed in this Pull Request

This adds the Stripe BNPL Payment Method Messaging Element to the cart summary on cart pages. For for classic cart, the PMME is mounted to a div that's added to the page. For the cart block, the Stripe `<PaymentMethodMessagingElement />` component is used. Using this element required updating the `stripe/react-stripe-js` package. I've used the latest version of the package, so one thing to consider is whether this update will cause any issues and if an earlier package should be used.

This also integrates the Appearance API based on #8421. 









#### Testing instructions

1. Enable BNPL methods, i.e. Affirm, After Pay, Klarna
2. Add a product to the cart
3. Ensure that the shopper currency, country, and total amounts align with the requirements for BNPL payment methods. Each set of requirements can be found in the payment methods class. For example, the Klarna requirements can be found [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/payment-methods/class-klarna-payment-method.php#L34-L102).
4. Check that the BNPL message is rendered on the following pages/scenarios and that the displayed amount updates when cart totals change(quantity change or coupons): 

- Classic theme using cart shortcode and cart block
- Block based theme using cart shortcode(cart block will be displayed) and cart block
- Product page

**Testing the Appearance API**
For classic themes, add some custom CSS to change fonts and background colors, such as: 
```
body { 
    background: #000; 
    color: #fff;
}
```

Closing the Customizer will clear the transient.

For block based themes, change to a dark style. In Twenty Twenty-Four, easy to go **Appearance > Editor  > Styles > Onyx**. Saving this style doesn't clear the transient, unfortunately. It will have to be cleared manually. `npm run wp transient delete wcpay_upe_bnpl_cart_block_appearance` and `npm run wp transient delete wcpay_upe_bnpl_cart_block_appearance_theme` should do the trick.





-------------------


**Post merge**
